polyfill/lib/ecmascript.mjs

```diff
@@ -3190,13 +3218,29 @@ function NudgeToCalendarUnit(
   // Create a duration with smallestUnit trunc'd towards zero
   // Create a separate duration that incorporates roundingIncrement
   let r1, r2, startDuration, endDuration;
+  var didExpandCalendarUnit = false;
+  const compare = (d1, d2) => CompareDurations(d1, d2, undefined, ToTemporalDate(isoDateTime.isoDate), unit, unit);
+  var cmpResult = 0;
   switch (unit) {
     case 'year': {
       const years = RoundNumberToIncrement(duration.date.years, increment, 'trunc');
       r1 = years;
       r2 = years + increment * sign;
       startDuration = { years: r1, months: 0, weeks: 0, days: 0 };
       endDuration = { ...startDuration, years: r2 };
+      cmpResult = compare(CombineDateAndTimeDuration(endDuration, TimeDuration.ZERO), duration);
+      if ((sign > 0 && cmpResult != 1) || (sign < 0 && cmpResult != -1)) {
+        didExpandCalendarUnit = true;
+        r1 = r2;
+        r2 = years + increment * 2 * sign;
```

Comment:
Why is the ×2 necessary here and in the months case below? Previously r1 and r2 would be exactly one rounding increment apart.

Reply:
Because `r1` after assigning `r2` to it is now equal to `years + increment * sign` -- so the `*2` is so that `r1` and `r2` will still be exactly one rounding increment apart.